components/wallets/pkg/manager.go

```diff
@@ -123,25 +125,35 @@ func (m *Manager) Debit(ctx context.Context, debit Debit) (*DebitHold, error) {
 		}
 	}
 
-	script := sdk.Script{
-		Plain: BuildDebitWalletScript(sources...),
-		Vars: map[string]interface{}{
-			"destination": dest.getAccount(m.chart),
-			"amount": map[string]any{
-				// @todo: upgrade this to proper int after sdk is updated
-				"amount": debit.Amount.Amount.Uint64(),
-				"asset":  debit.Amount.Asset,
+	postTransaction := sdk.PostTransaction{
+		Script: &sdk.PostTransactionScript{
+			Plain: BuildDebitWalletScript(sources...),
+			Vars: map[string]interface{}{
+				"destination": dest.getAccount(m.chart),
+				"amount": map[string]any{
+					// @todo: upgrade this to proper int after sdk is updated
+					"amount": debit.Amount.Amount.Uint64(),
+					"asset":  debit.Amount.Asset,
+				},
 			},
 		},
 		Metadata: TransactionMetadata(debit.Metadata),
 		//nolint:godox
 		// TODO: Add set account metadata for hold when released on ledger (v1.9)
 	}
+
 	if debit.Reference != "" {
-		script.Reference = &debit.Reference
+		postTransaction.Reference = &debit.Reference
+	}
+
+	if _, err := m.client.CreateTransaction(ctx, m.ledgerName, postTransaction); err != nil {
+		if machine.IsScriptErrorWithCode(err, apierrors.ErrInsufficientFund) {
```

Comment:
I don't think this can work.
The error return by the SDK is not of type *machine.ScriptError{}.
You can compare the code, but not the structure.